tests/jerry/es2015/regression-test-issue-2544.js

```diff
@@ -13,4 +13,8 @@
 // limitations under the License.
 
 Object.defineProperty(Array.prototype, 0, { get : function () { throw $; } });
-Promise.race([ , this]).then(Error);
+var asyncPassed = false;
+Promise.race([ , this]).then(Error).catch(function(err) { asyncPassed = (err instanceof ReferenceError); });
+new Promise(function() {
+  throw 5;
+}).then(Error).catch(function() { assert(asyncPassed); });
```

Comment:
Is the Promise execution/resolve order guaranteed in this test case? Could it happen that the race finishes after the Promise which is after it?

Reply:
It is guaranteed to happen in this order.